code/core/src/manager/components/TourGuide/TourGuide.tsx

```diff
@@ -0,0 +1,200 @@
+import type { ComponentProps } from 'react';
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { createRoot } from 'react-dom/client';
+
+import { darken } from 'polished';
+import type { CallBackProps } from 'react-joyride';
+import Joyride, { ACTIONS, type Step } from 'react-joyride';
+import { useTheme } from 'storybook/theming';
+import { ThemeProvider, convert, themes } from 'storybook/theming';
+
+import { HighlightElement } from './HighlightElement';
+import { TourTooltip } from './TourTooltip';
+
+type StepDefinition = {
+  key?: string;
+  highlight?: string;
+  hideNextButton?: boolean;
+  onNext?: ({ next }: { next: () => void }) => void;
+} & Partial<
+  Pick<
+    // Unfortunately we can't use ts-expect-error here for some reason
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore Ignore circular reference
```

Comment:
This `ts-ignore` here, will haunt you again later down the line. 
What is the cause of this, we need to get to the bottom of it.